pkg/openstack/loadbalancer_test.go

```diff
@@ -1108,3 +1108,130 @@ func Test_getStringFromServiceAnnotation(t *testing.T) {
+				{
+					Type:    corev1.NodeInternalIP,
+					Address: "192.168.1.1",
+				},
+			},
+		},
+	}
+	node2 := &corev1.Node{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "node-2",
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{
+				{
+					Type:    corev1.NodeInternalIP,
+					Address: "192.168.1.2",
+				},
+			},
+		},
+	}
+
+	// Instantiate an object of your LbaasV2 struct
+	lbaas := &LbaasV2{}
+
+	testCases := []struct {
+		name        string
+		nodes       []*corev1.Node
+		port        corev1.ServicePort
+		svcConf     *serviceConfig
+		expectedLen int
+		expectError bool
+	}{
+		{
+			name:  "NodePortequalszero",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 0},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+			},
+			expectedLen: 0,
+			expectError: false,
+		},
+		{
+			name:  "ErrNoAddressFound happens for all nodes",
+			nodes: []*corev1.Node{},
+			port:  corev1.ServicePort{NodePort: 0},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+			},
+			expectedLen: 0,
+			expectError: false,
+		},
+		{
+			name:  "Valid nodes, canUseHTTPMonitor=false",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 8080},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+				enableMonitor:       false,
+			},
+			expectedLen: 2,
+			expectError: false,
+		},
+		{
+			name:  "Valid nodes, canUseHTTPMonitor=true",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 8080},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+				enableMonitor:       true,
+			},
+			expectedLen: 2,
+			expectError: false,
+		},
+		{
+			name:  "Invalid preferred IP family, fallback to default",
+			nodes: []*corev1.Node{node1, node2},
+			port:  corev1.ServicePort{NodePort: 0},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   "invalid-family",
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+			},
+			expectedLen: 0,
+			expectError: false,
+		},
+		{
+			name:  "ErrNoAddressFound happens and a member is ignored",
+			nodes: []*corev1.Node{},
+			port:  corev1.ServicePort{NodePort: 8080},
+			svcConf: &serviceConfig{
+				preferredIPFamily:   corev1.IPv4Protocol,
+				lbMemberSubnetID:    "subnet-12345-test",
+				healthCheckNodePort: 8081,
+				enableMonitor:       false,
+			},
+			expectedLen: 0,
+			expectError: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			members, _, err := lbaas.buildBatchUpdateMemberOpts(tc.port, tc.nodes, tc.svcConf)
```

Comment:
Do you want to also write some assertions regarding the second returned value (`newMembers`)?

Reply:
Yes. I would do that right away. Thanks

Reply:
@pierreprinetti Please you can check it out again